src/gpt_review/_ask.py

```diff
@@ -20,9 +21,36 @@
 DEFAULT_KEY_VAULT = "https://dciborow-openai.vault.azure.net/"
 
 
-def _ask(question, max_tokens=100):
+def validate_parameter_range(namespace):
+    """Validate that max_tokens is in [1,4000], temperature and top_p are in [0,1], and
+    frequency_penalty and presence_penalty are in [0,2]"""
+    if namespace.max_tokens is not None:
+        if namespace.max_tokens < 1 or namespace.max_tokens > 4000:
+            raise CLIError("--max_tokens must be an integer between 1 and 4000")
+    if namespace.temperature is not None:
+        if namespace.temperature < 0 or namespace.temperature > 1:
+            raise CLIError("--temperature must be a float between 0 and 1")
+    if namespace.top_p is not None:
+        if namespace.top_p < 0 or namespace.top_p > 1:
+            raise CLIError("--top-p must be a float between 0 and 1")
+    if namespace.frequency_penalty is not None:
+        if namespace.frequency_penalty < 0 or namespace.frequency_penalty > 2:
+            raise CLIError("--frequency-penalty must be a float between 0 and 2")
+    if namespace.presence_penalty is not None:
+        if namespace.presence_penalty < 0 or namespace.presence_penalty > 2:
+            raise CLIError("--presence-penalty must be a float between 0 and 2")
```

Comment:
```suggestion
def validate_parameter_range(namespace):
    """Validate that max_tokens is in [1,4000], temperature and top_p are in [0,1], and frequency_penalty and presence_penalty are in [0,2]"""
    _range_validation(namespace.max_tokens, "max-tokens", 1, 4000)
    _range_validation(namespace.temperature, "temperature", 0, 1)
    _range_validation(namespace.top_p, "top-p", 0, 1)
    _range_validation(namespace.frequency_penalty, "frequency-penalty", 0, 2)
    _range_validation(namespace.presence_penalty, "presence-penalty", 0, 2)


def _range_validation(param, name, min_value, max_value):
    if param and (param < min_value or param > max_value):
        raise CLIError("--%s must be a %s greater than %s and %s", name, type(param), min_value, max_value)

```

Here is how we can factor out the duplicated logic. Also, `if namespace.presence_penalty is not None` should be equivalent to `if namespace.presence_penalty`

Reply:
and lets mark method as private by appending "_", so we let people know not to depend on this method.

Reply:
and lets make the method private by appending "_" so we let people know not to rely on this.